### ISSUE TITLE
os-acme-client 4.0 : DNS-01 nsupdate: Help for secret key should mention format

### ISSUE THREAD

**Issue by phloggu** [opnsense/plugins#3799]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

- [X] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [X] I have searched the existing issues, open and closed, and I'm convinced that mine is new.
- [X] When the request is meant for an existing plugin, I've added its name to the title.

**Is your feature request related to a problem? Please describe.**
It was unclear for me what to enter in the text field "Secret Key" for the challenge type "DNS-01", DNS Service "nsupdate (RFC 2136)". The help text currently says:

_Requires the whole key file in a format that is compatible with nsupdate._ 

I entered the content of the key file as produced by `tsig-keygen` resp. `dnssec-keygen`:

```
Private-key-format: v1.3
Algorithm: 165 (HMAC_SHA512)
Key: 7DXXX...
Bits: AAA=
Created: 20240205213656
Publish: 20240205213656
Activate: 20240205213656
```

...which did not work.

After lengthy researches I found that the form, which is used in `named.conf` to define keys, did work:

```
key "_acme-challenge.example.com." {
  algorithm hmac-sha512;
  secret "7DXXX...";
};
```

The help text did not help me much. That's why I am asking to change it. 

I don't know whether the code could work with the contents of a key file from `tsig-keygen` (I think it would need the pair of files K\<xxx\>.key and K\<xxx\>.private), if yes, then this request to change the help text is almost obsolete and there should be some testing with that key file format or describe how to use it. 

**Describe the solution you'd like**
Since `nsupdate` supports two formats for key files, the help text is not wrong, but could be more elaborate to help. Please change the help text to say _content_ of the key file, not to confuse with a path to a key file. Secondly, for as long as it is unclear whether the key file format from  `tsig-keygen` does work or not, mention the format that works:

_Enter the the key exactly as it is defined in named.conf of your bind server configuration, use the whole key{}-statement._

Please change this if needed, I am not a native speaker.

**Describe alternatives you've considered**
Alternatively:
_Requires the whole content of the key file in a format that is compatible with nsupdate, which is one of two formats:  a single key statement as used in named.conf of the bind server software, which was generated by ddns-confgen; or the content of the key file generated by dnssec-keygen resp. tsig-keygen._

**Additional context**
It might be possible that even SIG(0) key pairs are supported. In this case the help text should mention this as well.

**Comment by fraenki** [comment]
Yes, it's pretty cumbersome that one has to specify it in named.conf format. I'll consider an improvement for one of the next releases.

**Comment by phloggu** [comment]
I don't feel this to be cumbersome: I have to add it anyway to the name server config, so it's easy to paste it to the plugin just afterwards. The only thing is: one has to know about it.
In case the one using the key and the one defining it on the server is not the same person, it looks different though. If you come up with a good solution, that's great, if you have other things to do first, changing the help text would be good for now. Thank you!

**Comment by noseshimself** [comment]
A better alternative might be adding another input field for nsupdate queries and splitting it into "key name" and "key"; I'm using Technitium DNS and it would be even more cumbersome for me to hand-craft an nsupdate-acceptable file from those two things I can cut&paste from it.

**Comment by InvoluntaryG17HubUser** [comment]
Hey just chiming in on this issue.

I just ran into this myself. Updating the help text would have saved me half an hour of trial and error.

I am find with it being a bind specific format, however I wish that this will be made clean in the help text.
